core/src/main/scala/com/databricks/labs/remorph/parsers/SqlErrorStrategy.scala

```diff
@@ -15,6 +21,53 @@ import org.antlr.v4.runtime.misc.IntervalSet
  */
 abstract class SqlErrorStrategy extends DefaultErrorStrategy {
 
+  @throws[RecognitionException]
+  override def sync(recognizer: Parser): Unit = {
+    val tokens: TokenStream = recognizer.getInputStream
+    val startIndex: Int = tokens.index
+    val skippedTokens: util.List[Token] = new util.ArrayList[Token]
+    try {
+      super.sync(recognizer)
+    } catch {
+      case e: RecognitionException => throw e // Throw back to parser
+    } finally {
+      val endIndex: Int = tokens.index
+      for (i <- startIndex until endIndex) {
+        skippedTokens.add(tokens.get(i))
+      }
+      if (!skippedTokens.isEmpty) {
+        val firstToken: Token = skippedTokens.get(0)
+        val lastToken: Token = skippedTokens.get(skippedTokens.size - 1)
+        val errorToken: CommonToken = new CommonToken(
+          new Pair(firstToken.getTokenSource, firstToken.getInputStream),
+          Token.INVALID_TYPE,
+          Token.DEFAULT_CHANNEL,
+          firstToken.getStartIndex,
+          lastToken.getStopIndex)
+        errorToken.setLine(firstToken.getLine)
+        errorToken.setCharPositionInLine(firstToken.getCharPositionInLine)
+        val errorNode = new RemorphErrorNode(errorToken)
+        for (token <- skippedTokens.asScala) {
+          errorNode.addChild(new TerminalNodeImpl(token))
+        }
```

Comment:
It is strictly equivalent, but the suggested version makes it more obvious that the expression is side-effecting

```suggestion
        skippedTokens.asScala.foreach { token =>
          errorNode.addChild(new TerminalNodeImpl(token))
        }
```

Reply:
In fact I have changed my mind on accumulating the text this way and simplified the ErrorNode, so this code is no longer present.